.github/workflows/check-submodule-pointers.yml

```diff
@@ -0,0 +1,38 @@
+name: Check submodule pointers
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  check-submodules:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Check ql submodule
+        env: 
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          # Tip of the lgtm.com branch
+          target_sha=$(gh api repos/github/codeql/git/ref/heads/lgtm.com --jq '.object.sha')
+          
+          if [ $(git rev-parse @:./ql) == $target_sha ]; then
+            echo "The ql submodule already points to the tip of the lgtm.com branch. Nothing to do."
+          else
+            cd ql
+            git fetch origin $target_sha
+            git checkout $target_sha
+            cd ..
+            git add ql
+            git config user.name "github-actions[bot]"
+            git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git commit -m "Bump ql submodule to latest lgtm.com"
+            git push origin $GITHUB_REF
```

Comment:
`$GITHUB_REF` may not be quite right, since it refers to "refs/pull/NUMBER/merge" in the case of a PR. Possibly need to switch to `$GITHUB_HEAD_REF` when triggered on PRs.

Reply:
Passing observation: you can keep this task simpler and just have the workflow perform a check, returning a non-zero exit code and an error message if the pointers don't line up. No need to actually bump if they're mismatched. We can make a separate workflow to automate that if we want.